### PR TITLE
ISSD-2051 [PRM] Allow more than one 3rd party invoice for MDFs

### PR DESCRIPTION
**Pull request by eliasantosf**
Task: [ISSD-2051](https://liferay.atlassian.net/browse/ISSD-2051)

### CODE REVIEW

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/PRMForm/components/fields/InputMultipleFiles/InputMultipleFiles.tsx [hunk 1]
```diff
@@ -10,13 +10,15 @@ import {useDropzone} from 'react-dropzone';
...
 import MDFClaim from '../../../../../interfaces/mdfClaim';
+import {ResourceName} from '../../../../../services/liferay/object/enum/resourceName';
 import PRMFormik from '../../../../PRMFormik';
 import ListFiles from '../InputMultipleFilesListing/components/ListFiles';
 import PRMFormFieldStateProps from '../common/interfaces/prmFormFieldStateProps';
 import PRMFormMultipleFilesProps from '../common/interfaces/prmFormMultipleFilesProps';
 
 interface IProps {
 	acceptedFilesExtensions: string;
+	fileResourseName: ResourceName;
```
Comment (kylembischof) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/PRMForm/components/fields/InputMultipleFiles/InputMultipleFiles.tsx:21]: `fileResourseName` spelled incorrectly. should be `fileResourceName`. also fix all the other places the name is used

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/interfaces/liferayFile.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@
  */
 
 export default interface LiferayFile extends Partial<File> {
-	activityDocumentId?: number;
+	claimDocumentId?: number;
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/interfaces/liferayFile.ts:7]:
```suggestion
	objectId?: number;
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/PRMForm/components/fields/InputMultipleFiles/InputMultipleFiles.tsx [hunk 3]
```diff
@@ -25,6 +27,7 @@ const InputMultipleFiles = ({
 	acceptedFilesExtensions,
 	description,
 	field,
+	fileResourseName,
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/PRMForm/components/fields/InputMultipleFiles/InputMultipleFiles.tsx:30]:
```suggestion
	resourceName,
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/PRMForm/components/fields/InputMultipleFilesListing/components/ListFiles/ListFiles.tsx [hunk 4]
```diff
@@ -15,14 +15,15 @@ import {ResourceName} from '../../../../../../../services/liferay/object/enum/re
 
 interface IProps {
 	arrayHelpers: ArrayHelpers;
+	fileResourseName: ResourceName;
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/PRMForm/components/fields/InputMultipleFilesListing/components/ListFiles/ListFiles.tsx:18]:
```suggestion
	resourceName: ResourceName;
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/PRMForm/components/fields/InputMultipleFilesListing/components/ListFiles/ListFiles.tsx [hunk 5]
```diff
@@ -64,10 +65,10 @@ const ListFiles = ({arrayHelpers, files, meta}: IProps) => {
 									arrayHelpers.remove(index);
 								}
 
-								if (file.activityDocumentId) {
+								if (file.claimDocumentId) {
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/PRMForm/components/fields/InputMultipleFilesListing/components/ListFiles/ListFiles.tsx:68]:
```suggestion
								if (file.objectId) {
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/PRMForm/components/fields/InputMultipleFilesListing/components/ListFiles/ListFiles.tsx [hunk 6]
```diff
@@ -64,10 +65,10 @@ const ListFiles = ({arrayHelpers, files, meta}: IProps) => {
...
 									await deleteObjectEntry(
-										ResourceName.MDF_CLAIM_ACTIVITY_DOCUMENTS,
-										file.activityDocumentId
+										fileResourseName,
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/PRMForm/components/fields/InputMultipleFilesListing/components/ListFiles/ListFiles.tsx:70]:
```suggestion
										resourceName,
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/PRMForm/components/fields/InputMultipleFilesListing/components/ListFiles/ListFiles.tsx [hunk 7]
```diff
@@ -64,10 +65,10 @@ const ListFiles = ({arrayHelpers, files, meta}: IProps) => {
...
+										file.claimDocumentId
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/PRMForm/components/fields/InputMultipleFilesListing/components/ListFiles/ListFiles.tsx:71]:
```suggestion
										file.objectId
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/PRMForm/components/fields/InputMultipleFilesListing/InputMultipleFilesListing.tsx [hunk 8]
```diff
@@ -5,12 +5,14 @@
...
 import PRMForm from '../../..';
 import LiferayFile from '../../../../../interfaces/liferayFile';
+import {ResourceName} from '../../../../../services/liferay/object/enum/resourceName';
 import PRMFormik from '../../../../PRMFormik';
 import PRMFormFieldProps from '../common/interfaces/prmFormFieldProps';
 
 interface IProps {
 	acceptedFilesExtensions: string;
 	description: string;
+	fileResourseName: ResourceName;
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/PRMForm/components/fields/InputMultipleFilesListing/InputMultipleFilesListing.tsx:15]:
```suggestion
	resourceName: ResourceName;
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/PRMForm/components/fields/InputMultipleFilesListing/InputMultipleFilesListing.tsx [hunk 9]
```diff
@@ -20,6 +22,7 @@ interface IProps {
 const InputMultipleFilesListing = ({
 	acceptedFilesExtensions,
 	description,
+	fileResourseName,
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/PRMForm/components/fields/InputMultipleFilesListing/InputMultipleFilesListing.tsx:25]:
```suggestion
	resourceName,
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/PRMForm/components/fields/InputMultipleFilesListing/InputMultipleFilesListing.tsx [hunk 10]
```diff
@@ -30,6 +33,7 @@ const InputMultipleFilesListing = ({
 		acceptedFilesExtensions={acceptedFilesExtensions}
 		component={PRMForm.InputMultipleFiles}
 		description={description}
+		fileResourseName={fileResourseName}
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/PRMForm/components/fields/InputMultipleFilesListing/InputMultipleFilesListing.tsx:36]:
```suggestion
		resourceName={resourceName}
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/services/liferay/object/mdf-claim-documents/createMDFClaimDocument.ts [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+import {LiferayAPIs} from '../../common/enums/apis';
+import liferayFetcher from '../../common/utils/fetcher';
+
+export default async function createMDFClaimDocument(
+	mdfClaimADocumentDTO: LiferayFile[]
+) {
+	return await liferayFetcher.post(
+		`/o/${LiferayAPIs.OBJECT}/mdfclaimdocuments/batch`,
+		Liferay.authToken,
+		mdfClaimADocumentDTO
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/services/liferay/object/mdf-claim-documents/createMDFClaimDocument.ts:17]:
```suggestion
		mdfClaimDocumentDTOs
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/services/liferay/object/mdf-claim-documents/createMDFClaimDocument.ts [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Liferay} from '../..';
+import LiferayFile from '../../../../interfaces/liferayFile';
+import {LiferayAPIs} from '../../common/enums/apis';
+import liferayFetcher from '../../common/utils/fetcher';
+
+export default async function createMDFClaimDocument(
+	mdfClaimADocumentDTO: LiferayFile[]
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/services/liferay/object/mdf-claim-documents/createMDFClaimDocument.ts:12]:
```suggestion
	mdfClaimDocumentDTOs: LiferayFile[]
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/services/liferay/object/mdf-claim-documents/updateMDFClaimDocument.ts [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Liferay} from '../..';
+import LiferayFile from '../../../../interfaces/liferayFile';
+import {LiferayAPIs} from '../../common/enums/apis';
+import liferayFetcher from '../../common/utils/fetcher';
+
+export default async function updateMDFClaimDocument(
+	mdfClaimDocumentDTO: LiferayFile[]
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/services/liferay/object/mdf-claim-documents/updateMDFClaimDocument.ts:12]:
```suggestion
	mdfClaimDocumentDTOs: LiferayFile[]
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/services/liferay/object/mdf-claim-documents/updateMDFClaimDocument.ts [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+) {
+	return await liferayFetcher.put(
+		`/o/${LiferayAPIs.OBJECT}/mdfclaimdocuments/batch`,
+		Liferay.authToken,
+		mdfClaimDocumentDTO
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/services/liferay/object/mdf-claim-documents/updateMDFClaimDocument.ts:17]:
```suggestion
		mdfClaimDocumentDTOs
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/utils/dto/mdf-claim-activity-document/getDocumentDTOFromLiferayFile.ts [hunk 15]
```diff
@@ -14,7 +14,7 @@ export default function getDocumentDTOFromLiferayFile(
 	dtoMDFClaimActivityId: number
 ): MDFClaimActivityDocumentDTO {
 	return {
-		id: liferayFile.activityDocumentId,
+		id: liferayFile.claimDocumentId,
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/utils/dto/mdf-claim-activity-document/getDocumentDTOFromLiferayFile.ts:17]:
```suggestion
		id: liferayFile.objectId,
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/utils/getPOPDocument.ts [hunk 16]
```diff
@@ -11,7 +11,7 @@ export function getPOPDocument(
 	mdfClaimActivityDocumentDTO: MDFClaimActivityDocumentDTO
 ): LiferayFile {
 	return {
-		activityDocumentId: mdfClaimActivityDocumentDTO.id,
+		claimDocumentId: mdfClaimActivityDocumentDTO.id,
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/utils/getPOPDocument.ts:14]:
```suggestion
		objectId: mdfClaimActivityDocumentDTO.id,
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/components/MDFClaimPage/components/ActivityClaimPanel/components/ContentMarketingPopFields/ContentMarketingPopFields.tsx [hunk 17]
```diff
@@ -33,6 +34,7 @@ const ContentMarketingPopFields = ({
 			<InputMultipleFilesListing
 				acceptedFilesExtensions="doc, docx, jpg, jpeg, png, tif, tiff, pdf"
 				description="Drag and drop your files here to upload."
+				fileResourseName={ResourceName.MDF_CLAIM_ACTIVITY_DOCUMENTS}
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/components/MDFClaimPage/components/ActivityClaimPanel/components/ContentMarketingPopFields/ContentMarketingPopFields.tsx:37]:
```suggestion
				resourceName={ResourceName.MDF_CLAIM_ACTIVITY_DOCUMENTS}
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/components/MDFClaimPage/components/ActivityClaimPanel/components/DigitalMarketingPopFields/DigitalMarketingPopFields.tsx [hunk 18]
```diff
@@ -36,6 +37,7 @@ const DigitalMarketingPopFields = ({
 			<InputMultipleFilesListing
 				acceptedFilesExtensions="doc, docx, jpg, jpeg, png, tif, tiff, pdf"
 				description="Drag and drop your files here to upload."
+				fileResourseName={ResourceName.MDF_CLAIM_ACTIVITY_DOCUMENTS}
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/components/MDFClaimPage/components/ActivityClaimPanel/components/DigitalMarketingPopFields/DigitalMarketingPopFields.tsx:40]:
```suggestion
				resourceName={ResourceName.MDF_CLAIM_ACTIVITY_DOCUMENTS}
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/components/MDFClaimPage/components/ActivityClaimPanel/components/EventPopFields/EventPopFields.tsx [hunk 19]
```diff
@@ -48,6 +49,7 @@ const EventPopFields = ({
 			<InputMultipleFilesListing
 				acceptedFilesExtensions="doc, docx, jpg, jpeg, png, tif, tiff, pdf"
 				description="Drag and drop your files here to upload."
+				fileResourseName={ResourceName.MDF_CLAIM_ACTIVITY_DOCUMENTS}
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/components/MDFClaimPage/components/ActivityClaimPanel/components/EventPopFields/EventPopFields.tsx:52]:
```suggestion
				resourceName={ResourceName.MDF_CLAIM_ACTIVITY_DOCUMENTS}
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/components/MDFClaimPage/components/ActivityClaimPanel/components/EventPopFields/EventPopFields.tsx [hunk 20]
```diff
@@ -67,6 +69,7 @@ const EventPopFields = ({
...
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/components/MDFClaimPage/components/ActivityClaimPanel/components/EventPopFields/EventPopFields.tsx:72]:
```suggestion
				resourceName={ResourceName.MDF_CLAIM_ACTIVITY_DOCUMENTS}
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/components/MDFClaimPage/components/ActivityClaimPanel/components/EventPopFields/EventPopFields.tsx [hunk 21]
```diff
@@ -86,6 +89,7 @@ const EventPopFields = ({
...
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/components/MDFClaimPage/components/ActivityClaimPanel/components/EventPopFields/EventPopFields.tsx:92]:
```suggestion
				resourceName={ResourceName.MDF_CLAIM_ACTIVITY_DOCUMENTS}
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/components/MDFClaimPage/components/ActivityClaimPanel/components/MiscellaneousMarketingPopFields/MiscellaneousMarketingPopFields.tsx [hunk 22]
```diff
@@ -57,6 +58,7 @@ const MiscellaneousMarketingPopFields = ({
 			<InputMultipleFilesListing
 				acceptedFilesExtensions="jpg, jpeg, png, tif, tiff, pdf"
 				description="Drag and drop your files here to upload."
+				fileResourseName={ResourceName.MDF_CLAIM_ACTIVITY_DOCUMENTS}
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/components/MDFClaimPage/components/ActivityClaimPanel/components/MiscellaneousMarketingPopFields/MiscellaneousMarketingPopFields.tsx:61]:
```suggestion
				resourceName={ResourceName.MDF_CLAIM_ACTIVITY_DOCUMENTS}
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/components/MDFClaimPage/components/ActivityClaimPanel/components/MiscellaneousMarketingPopFields/MiscellaneousMarketingPopFields.tsx [hunk 23]
```diff
@@ -75,6 +77,7 @@ const MiscellaneousMarketingPopFields = ({
 			<InputMultipleFilesListing
 				acceptedFilesExtensions="doc, docx, jpg, jpeg, png, tif, tiff, pdf"
 				description="Drag and drop your files here to upload."
+				fileResourseName={ResourceName.MDF_CLAIM_ACTIVITY_DOCUMENTS}
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/components/MDFClaimPage/components/ActivityClaimPanel/components/MiscellaneousMarketingPopFields/MiscellaneousMarketingPopFields.tsx:80]:
```suggestion
				resourceName={ResourceName.MDF_CLAIM_ACTIVITY_DOCUMENTS}
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/components/MDFClaimPage/MDFClaimPage.tsx [hunk 24]
```diff
@@ -164,28 +165,28 @@ const MDFClaimPage = ({
...
-								);
-							}
-
-							setFieldValue(`reimbursementInvoice`, liferayFile);
-						}}
-						outline
+					<InputMultipleFilesListing
+						acceptedFilesExtensions="doc, docx, jpg, jpeg, png, tif, tiff, pdf"
+						description="Drag and drop your files here to upload an invoice for the Total Claim Amount."
+						fileResourseName={ResourceName.MDF_CLAIM_DOCUMENTS}
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/components/MDFClaimPage/MDFClaimPage.tsx:171]:
```suggestion
						resourceName={ResourceName.MDF_CLAIM_DOCUMENTS}
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/utils/checkRequiredListOfQualifiedLeads.ts [hunk 25]
```diff
@@ -11,8 +11,8 @@ export default function checkRequiredListOfQualifiedLeads(
 	typeActivity: LiferayPicklist
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/utils/checkRequiredListOfQualifiedLeads.ts:11]:
```suggestion
	typeActivity?: LiferayPicklist
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/hooks/useIsChannel.ts
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/hooks/useIsChannel.ts:1]: Undo this change.

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/interfaces/dto/mdfClaimDocumentDTO.ts [hunk 26]
```diff
@@ -0,0 +1,12 @@
...
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import LiferayFile from '../liferayFile';
+import MDFClaimDocument from '../mdfClaimDocument';
+
+export default interface MDFClaimDocumentDTO
+	extends Omit<MDFClaimDocument, 'reimbursementInvoices'> {
+	reimbursementInvoices?: LiferayFile & number;
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/interfaces/dto/mdfClaimDocumentDTO.ts:11]:
```suggestion
	file?: LiferayFile & number;
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/utils/getInvoiceFromMDFClmDocs.ts [hunk 27]
```diff
@@ -0,0 +1,31 @@
...
+				documentId: currentDocument.reimbursementInvoices?.id,
+				link: currentDocument.reimbursementInvoices?.link,
+				name:
+					currentDocument.reimbursementInvoices?.name &&
+					getNameFromMDFClaimDocument(
+						currentDocument.reimbursementInvoices.name
+					),
+			};
+
+			accumulatorDocuments?.push(reimbursementInvoiceFile);
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/utils/getInvoiceFromMDFClmDocs.ts:25]:
```suggestion
		        objectId: currentDocument.id,
				documentId: currentDocument.reimbursementInvoice?.id,
				link: currentDocument.reimbursementInvoice?.link,
				name:
					currentDocument.reimbursementInvoice?.name &&
					getNameFromMDFClaimDocument(
						currentDocument.reimbursementInvoice.name
					),
			};

			accumulatorDocuments?.push(reimbursementInvoiceFile);
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/utils/getInvoiceFromMDFClmDocs.ts [hunk 28]
```diff
@@ -0,0 +1,31 @@
...
+
+			return accumulatorDocuments;
+		},
+		[] as LiferayFile[] & number[]
+	) as MDFClaimDocument[];
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/utils/getInvoiceFromMDFClmDocs.ts:30]: Let's try to avoid this casting, try to use the type correctly.

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/interfaces/mdfClaimDocument.ts [hunk 29]
```diff
@@ -0,0 +1,11 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import LiferayFile from './liferayFile';
+import LiferayObject from './liferayObject';
+
+export default interface MDFClaimDocument extends Partial<LiferayObject> {
+	reimbursementInvoices?: LiferayFile[] & number[];
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/interfaces/mdfClaimDocument.ts:10]:
```suggestion
	file?: LiferayFile & number;
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/services/liferay/object/mdf-claim-documents/updateMDFClaimDocument.ts [hunk 30]
```diff
@@ -0,0 +1,19 @@
...
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Liferay} from '../..';
+import LiferayFile from '../../../../interfaces/liferayFile';
+import {LiferayAPIs} from '../../common/enums/apis';
+import liferayFetcher from '../../common/utils/fetcher';
+
+export default async function updateMDFClaimDocument(
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/services/liferay/object/mdf-claim-documents/updateMDFClaimDocument.ts:11]:
```suggestion
export default async function updateMDFClaimDocuments(
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/services/liferay/object/mdf-claim-documents/createMDFClaimDocument.ts [hunk 31]
```diff
@@ -0,0 +1,19 @@
...
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Liferay} from '../..';
+import LiferayFile from '../../../../interfaces/liferayFile';
+import {LiferayAPIs} from '../../common/enums/apis';
+import liferayFetcher from '../../common/utils/fetcher';
+
+export default async function createMDFClaimDocument(
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/services/liferay/object/mdf-claim-documents/createMDFClaimDocument.ts:11]: Rename the file too
```suggestion
export default async function createMDFClaimDocuments(
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/utils/submitMDFClaimDocuments.ts [hunk 32]
```diff
@@ -0,0 +1,45 @@
...
+import createMDFClaimDocument from '../../../common/services/liferay/object/mdf-claim-documents/createMDFClaimDocument';
+import updateMDFClaimDocument from '../../../common/services/liferay/object/mdf-claim-documents/updateMDFClaimDocument';
+
+const submitMDFClaimDocuments = async (
+	reimbursementInvoices: LiferayFile[] & MDFClaimDocument[],
+	companyId: number,
+	dtoMDFClaimId: number
+) => {
+	const dtoMDFClaimDocumentsCreate: LiferayFile[] & number[] = [];
+	const dtoMDFClaimDocumentsUpdate: LiferayFile[] & number[] = [];
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/utils/submitMDFClaimDocuments.ts:17]:
```suggestion
	const dtoMDFClaimDocumentsCreate: LiferayFile[] = [];
	const dtoMDFClaimDocumentsUpdate: LiferayFile[] = [];
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/utils/submitMDFClaimDocuments.ts [hunk 33]
```diff
@@ -0,0 +1,45 @@
...
+
+	if (reimbursementInvoices?.length) {
+		reimbursementInvoices.map(async (reimbursementInvoice) => {
+			if (reimbursementInvoice.documentId) {
+				const dtoMDFClaimDocument = {
+					id: reimbursementInvoice.claimDocumentId,
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimForm/utils/submitMDFClaimDocuments.ts:23]:
```suggestion
					id: reimbursementInvoice.objectId,
```

**Review by jairmedeiros — CHANGES_REQUESTED** [review]

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/PRMForm/components/fields/InputMultipleFiles/InputMultipleFiles.tsx (hunk 1 above)
Comment (eliasantosf) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/components/PRMForm/components/fields/InputMultipleFiles/InputMultipleFiles.tsx:21]: Done!

**Comment by eliasantosf** [comment]
Changes made.

**Review by jairmedeiros — CHANGES_REQUESTED** [review]
Still have changes to do.

**Review by jairmedeiros — CHANGES_REQUESTED** [review]
Simple changes

**Review by jairmedeiros — CHANGES_REQUESTED** [review]
Still have changes to do.

**Review by jairmedeiros — APPROVED** [review]

**Comment by jairmedeiros** [comment]
@thaynam, could you please take a look too?

**Review by thaynam — APPROVED** [review]
@jairmedeiros works fine. thank you @eliasantosf

**Comment by jairmedeiros** [comment]
ci:forward:force

**Comment by jairmedeiros** [comment]
ci:forward:force

**Comment by thaynam** [comment]
[resent](https://github.com/liferay-customer-delivery/liferay-portal/pull/34)
